quinn-proto/src/connection.rs

```diff
@@ -630,42 +624,50 @@ impl Connection {
         let rtt = cmp::max(self.latest_rtt, self.smoothed_rtt);
         let loss_delay = rtt + ((rtt * self.config.time_threshold as u64) >> 16);
         let lost_send_time = now.saturating_sub(loss_delay);
-        let lost_pn = self
-            .largest_acked_packet
-            .saturating_sub(self.config.packet_threshold as u64);
-        for (&packet, info) in self.sent_packets.range(0..self.largest_acked_packet) {
-            if info.time_sent <= lost_send_time || packet <= lost_pn {
-                lost_packets.push(packet);
-            } else if self.loss_time == 0 {
-                self.loss_time = info.time_sent + loss_delay;
-            } else {
-                self.loss_time = cmp::min(self.loss_time, info.time_sent + loss_delay);
-            }
-        }
 
-        // OnPacketsLost
-        if let Some(largest_lost) = lost_packets.last().cloned() {
-            let old_bytes_in_flight = self.bytes_in_flight;
-            let largest_lost_time = self.sent_packets[&largest_lost].time_sent;
-            for packet in lost_packets {
-                let info = self.sent_packets.remove(&packet).unwrap();
-                if !info.in_flight {
-                    continue;
-                }
-                self.bytes_in_flight -= info.size as u64;
-                self.lost_packets += 1;
-                if info.is_crypto_packet {
-                    self.crypto_pending += info.retransmits;
+        let mut lost_nonack = false;
```

Comment:
(I know this is not new here and don't know if this is literally from the spec, but I dislike the `nonack` name... could use an underscore to disambiguate how to read it.)